bpo-43574: Dont overallocate list literals (#24954)
python/cpython: Misc/NEWS.d/next/Core and Builtins/2021-03-21-18-51-30.bpo-43574.mteI-I.rst

> Restores previous list memory behavior where lists initialized from literals

Reviewer: ```suggestion
``list` objects don't overallocate when starting empty or are set to be empty. This effectively restores previous ``list`` memory behavior for lists initialized from literals.
```

Author: > ``list`

Minor typo - missing backtick in first \`\`list\`\`

Author: Added a commit to adopt this NEWS description (slightly modified)

``list`` objects don't overallocate when starting empty and then extended, or

> when set to be empty.  This effectively restores previous ``list`` memory

Reviewer 2: quantify what "previous" means.  3.8 and earlier I believe.
